pilot/pkg/networking/grpcgen/grpcecho_test.go

```diff
@@ -0,0 +1,190 @@
+package grpcgen_test
+
+import (
+	"context"
+	"fmt"
+	"google.golang.org/grpc"
+	"istio.io/istio/pkg/test/echo/proto"
+	"math"
+	"net"
+	"testing"
+
+	//  To install the xds resolvers and balancers.
+	_ "google.golang.org/grpc/xds"
+
+	networking "istio.io/api/networking/v1alpha3"
+	"istio.io/istio/pilot/pkg/xds"
+	"istio.io/istio/pkg/config"
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/config/schema/collections"
+	"istio.io/istio/pkg/test/echo/client"
+	"istio.io/istio/pkg/test/echo/common"
+	"istio.io/istio/pkg/test/echo/server/endpoint"
+)
+
+const grpcEchoPort = 14058
+
+type echoCfg struct {
+	version   string
+	namespace string
+}
+
+type configGenTest struct {
+	*testing.T
+	endpoints []endpoint.Instance
+	ds        *xds.FakeDiscoveryServer
+}
+
+// newConfigGenTest creates a FakeDiscoveryServer that listens for gRPC on grpcXdsAddr
+// For each of the given servers, we serve echo (only supporting Echo, no ForwardEcho) and
+// create a corresponding WorkloadEntry. The WorkloadEntry will have the given format:
+//
+//    meta:
+//      name: echo-{generated portnum}-{server.version}
+//      namespace: {server.namespace or "default"}
+//      labels: {"app": "grpc", "version": "{server.version}"}
+//    spec:
+//      address: {grpcEchoHost}
+//      ports:
+//        grpc: {generated portnum}
+func newConfigGenTest(t *testing.T, discoveryOpts xds.FakeOptions, servers ...echoCfg) *configGenTest {
+	cgt := &configGenTest{T: t}
+	for i, s := range servers {
+		// TODO this breaks without extra ifonfig aliases on OSX, and probably elsewhere
```

Comment:
why does it break?

Reply:
is it plausible to skip if it is? Would probably not be ideal to break tests for people